### ISSUE TITLE
Cypress no longer works on AWS CodeBuild

### ISSUE THREAD

**Issue by ffxsam** [cypress-io/cypress#1373]
Cypress 2.x no longer works on AWS CodeBuild. I get this error during the testing phase:

```
error cypress@2.0.3: The engine "node" is incompatible with this module. Expected version ">=8.2.1".
```

What's the recommended course of action here?

**Comment by bahmutov** [comment]
Hmm, we have upgraded to new Electron version, which is 8, so we moved our base image and CLI package to Node 8. Can you change the environment on AWS CodeBuild to be Node 8?

**Comment by ffxsam** [comment]
Unfortunately not. Not without some Docker hacking which is a bit beyond my skills. Out of the box, CodeBuild only goes up to Node.js v7.0.0.

**Comment by bahmutov** [comment]
Hmm, ok this is definitely wrong - our CLI package (that's what gets installed from NPM) should only require Node >= 4 but the installed package json for 2.0.3 has much stricter requirements
```
"engines": {
    "node": ">=8.2.1"
}
"version": "2.0.3",
  "warnings": [
    {
      "code": "ENOTSUP",
      "required": {
        "node": ">=8.2.1"
      },
      "pkgid": "cypress@2.0.3"
    },
    {
      "code": "ENOTSUP",
      "required": {
        "node": ">=8.2.1"
      },
      "pkgid": "cypress@2.0.3"
    }
  ]
```
this might be introduced during build process, I will investigate. 
@ffxsam does the error actually prevent the installation or is it a warning?

**Comment by ffxsam** [comment]
I'm not 100% sure, but let me check and get back to you.

**Comment by brian-mann** [comment]
Yeah weird... we'll look into it. The CLI's `package.json` only sets node `>=4.0.0` here.

https://github.com/cypress-io/cypress/blob/develop/cli/package.json#L10

**Comment by bahmutov** [comment]
Seems we are overwriting engines in the CLI package.json with engines from the root package. This is wrong. When testing we should set config value to error in this case (https://docs.npmjs.com/misc/config#engine-strict) and then fix the engines property.

**Comment by brian-mann** [comment]
We'll get a patch release out this weekend.

**Comment by brian-mann** [comment]
Released in [`2.0.4`](https://docs.cypress.io/guides/references/changelog.html#2-0-4).
